src/hooks.ts

```diff
@@ -11,3 +14,40 @@ export type OnUnknownFieldType = (info: {
   typeName: string
   fieldName: string
 }) => void
+
+export type OnUnknownArgName = (info: {
+  unknownArgNames: string[]
+  error: Error
+  typeName: string
+  fieldName: string
+}) => void
+
+type Hooks =
+  | OnUnknownFieldType
+  | OnUnknownFieldName
+  | OnUnknownArgName
+  | undefined
+
+export function registerHook<T extends Hooks>(
+  hook: T,
+  params: Parameters<Exclude<T, undefined>>,
+  message: string,
+  stage: 'build' | 'walk',
```

Comment:
Seems like a good type var to keep in some utils/types module? `type NexusTypeGenStage = ...`. 

Moreover, nexus should export this type. It doesn't?